src/shell/output.c

```diff
@@ -409,15 +409,17 @@ static int shell_output_file (struct shell_output *out)
 }
 
 static int shell_output_write_leader (struct shell_output *out,
+                                      const char *type,
                                       json_t *o,
                                       flux_msg_handler_t *mh) // may be NULL
 {
     bool eof = false;
     json_t *entry;
 
-    if (iodecode (o, NULL, NULL, NULL, NULL, &eof) < 0)
+    if (strcmp (type, "data") == 0
+        && iodecode (o, NULL, NULL, NULL, NULL, &eof) < 0)
         goto error;
```

Comment:
errno is not set if the strcmp condition is true but `shell_output_write_leader() ` is expected to set errno or `flux_respond_error()` will fail with EINVAL.

Reply:
Pretty sure you just made this a non-error.

Reply:
Yeah, iirc we don't need to separately check for `eof` anymore so this conditional could be dropped.